### ISSUE TITLE
Percentile function does not work on MariaDB

### ISSUE THREAD

**Issue by daviddsp** [metabase/metabase#34993]
### Describe the bug

We recently tested the custom percentile expression feature in Metabase version v0.47.5, utilizing MariaDB version Jammy (11.1). Upon executing the custom percentile expression from the graphical query builder, we encountered the following error:

`You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'AS 'P' FROM 'commits' GROUP BY STR_TO_DATE(CONCAT(DATE_FORMAT('commits'.'comm...' at line 2`

Upon reviewing the SQL structure provided by Metabase in the (View SQL) option, we observed the following result:

`SELECT
  PERCENTILE_CONT(0.9) within group (
    order by
      commits.additions
  ) AS P
FROM
  commits`

We found that the query lacked the OVER (partition by committed_date) AS 'P' clause. Including this clause, the revised query is as follows:

`SELECT
  PERCENTILE_CONT(0.9) within group (
    order by
      commits.additions
  ) OVER (partition by committed_date) AS P
FROM
  commits`

Executing this revised query successfully returns the percentile value. We suggest updating the SQL definition for the custom percentile expression to incorporate the **OVER** clause within its structure, to avoid similar issues in the future.

### To Reproduce

1. Go to 'Graphical query Builder'
3. Click on 'Select source database, custom expression and add percentile: Percentile([column], 0.9)'
4. Scroll down to 'Visualisation Button'
5. See error

### Expected behavior

That updating the SQL definition for the custom percentile expression to incorporate the **OVER** clause within its structure

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1",
    "os.name": "Linux",
    "os.version": "6.1.51-0-virt",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "16.0 (Debian 16.0-1.pgdg120+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.4"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-10-20",
      "tag": "v0.47.5",
      "branch": "?",
      "hash": "e7ee12e"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Highest

**Comment by paoliniluis** [comment]
Thank you! do you know if this is something new in MariaDB?

**Comment by paoliniluis** [comment]
@daviddsp see my comment above

**Comment by lbrdnk** [comment]
According to Mariadb [docs](https://mariadb.com/kb/en/aggregate-functions/) they do not support aggregate version of percentile. Only the window version is available and as of current implementation, we do not support use of window functions from query builder. I believe this is rather feature request.

Side note. When user enters `percentile` into custom expression interface, they are provided with percentile usage pop-up, which could be misleading. When user enters just `percen` however, no auto-completion is done and that's correct.

**Comment by paoliniluis** [comment]
Thank you @lbrdnk 😀

**Comment by lbrdnk** [comment]
@paoliniluis, I mean feature request for either supporting window functions (as opened in #9393) or for simulation of aggregate with use of window functions as I've attempted in #27166. Change in linked PR is however not trivial and imho it deserves rather feature than bug fix status, but that's not up to me :)

@daviddsp I believe that as of now, we support only use of aggregation functions from query builder. The doc links you've provided points to `percentile_disc` window function. For further explanation see [this](https://stackoverflow.com/questions/71333509/window-functions-vs-group-by) stackoverflow question.

**Comment by paoliniluis** [comment]
Same error happens on the "performance" dashboard in Metabase Analytics...

**Comment by FilmonK** [comment]
Can confirm findings of @paoliniluis 
Happens with Metabase Analytics where the application database is on MySQL 8.4 and MariaDB 11.

**Comment by paoliniluis** [comment]
Incredibly enough, this is the same issue as https://github.com/metabase/metabase/issues/38100

**Comment by rileythomp** [comment]
I don't think this is the same issue as https://github.com/metabase/metabase/issues/38100. Also I can't reproduce, I get `Unsupported function Percentile` with maria db and `Aggregations like Percentile are not allowed when building a custom expression` on postgres. So this seems to be more of a feature request than a bug.
